### PR TITLE
Jnlua v2.1

### PR DESCRIPTION
**Pull request by tnsc4502**
## Description
1. The imported JNLua, a native lua lib based on JNI, runs faster and uses less memory and even brings almost no lua-side bug. (My own test showed that the memory usage reduced from 3GB to 690MB, depending on the area you played on).

2. The scripts themselves worked very well (no bizarre error such as class too large ...etc.) if the ScriptLib was not mentioned (scene3_dummy_points.lua can be loaded if you can tolerate some loading time).

Please carefully read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md) before making any pull requests.
And, **Do not make a pull request to merge into stable unless it is a hotfix. Use the development branch instead.**
## Type of changes

- [ ] Bug fix
- [x] New feature 
- [x] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

**Review by ghost — CHANGES_REQUESTED** [review]
Do not merge the formatting changes please, there were problems with that commit earlier and I had to hard reset.

**Comment by tnsc4502** [comment]
@Melledy Sorry, I didn't notice the formatting branch had been rolled back, I will re-submit a new pr based on the current fixed branch.

**Comment by Birdulon** [comment]
You don't have to make a new PR, you can force push to your existing PR branch. You can also set this as a draft until it's ready to be reviewed for merging.

**Pull request closed without merging**
